src/apps/annotation-image/RightDrawer/RightDrawer.tsx

```diff
@@ -51,7 +52,7 @@ export const RightDrawer = (props: RightDrawerProps) => {
     }
   });
 
-  const [tab, setTab] = useState<'ANNOTATIONS' | 'NOTES'>('ANNOTATIONS');
+  const [tab, setTab] = useState<'ANNOTATIONS' | 'NOTES'>(props.defaultTab);
```

Comment:
This is now __initializing__ the state from the hash param. But it's not keeping the hash in sync for state changes. Ideally, I think we'd like to have some kind of hook that acts as a drop-in replacement for `useState`, but works through the hash. That would give us share-able URLs on the one hand, but would use be useful for another purpose. Some context:

The root components (`ImageAnnotationDesktop`, `TextAnnotationDesktop`) need to be aware of whether the right tab is open, and whether it shows the in 'ANNOTATION' tab. It needs this in order to decide whether it should use floating popups when annotations are selected.

Right now, the implementation is a spaghetti mess. Right-panel closed/open state is maintained at the top; tab change triggers a callback, which then updates another state var `usePopup`. 

Now we could build everything off this nice & new `useAnnotationViewUIState` hook, which we'd keep at the top, and could use as the source of truth for the popup. (And any other things that might come up in the future.) __However__, in order for this to work we'd need to keep this state in sync with the UI. I wonder what your thoughts are @dleadbetter and @lwjameson?

Reply:
@rsimon I like it.  We should take the opportunity to future-proof when we can, and I think this might simplify in the end, if only simplifying the ability to understand the code. Always good in Open Source.